modules/types/include/hephaestus/types/uuid_v4.h

```diff
@@ -65,4 +69,12 @@ constexpr auto UuidV4::createMax() -> UuidV4 {
   return { .high = std::numeric_limits<uint64_t>::max(), .low = std::numeric_limits<uint64_t>::max() };
 }
 
+constexpr auto UuidV4::isValid() const -> bool {
+  static constexpr auto NIL = createNil();
+  static constexpr auto MAX = createMax();
+
+  return (*this != NIL) && (*this != MAX) &&                               // Nil and Max UUIDs are not valid
+         (this->high & 0x000000000000F000ULL) == 0x0000000000004000ULL &&  // Version 4
+         (this->low & 0xC000000000000000ULL) == 0x8000000000000000ULL;     // RFC 9562 variant
```

Comment:
Nit: This makes the version/variant checks self-explanatory.

```suggestion
  constexpr uint64_t VERSION_MASK = 0x000000000000F000ULL;
  constexpr uint64_t VERSION_4    = 0x0000000000004000ULL;

  constexpr uint64_t VARIANT_MASK = 0xC000000000000000ULL;
  constexpr uint64_t VARIANT_RFC  = 0x8000000000000000ULL;

  return (*this != NIL) && (*this != MAX) &&
         ((this->high & VERSION_MASK) == VERSION_4) &&
         ((this->low  & VARIANT_MASK) == VARIANT_RFC);
```